src/System.IO/tests/TextReader/TextReaderTests.cs

```diff
@@ -70,6 +70,18 @@ public void TestRead()
             }
         }
 
+        [Fact]
+        public void ReadZeroCharcters()
+        {
+            if (!PlatformDetection.IsFullFramework)
+            {
+                using (CharArrayTextReader tr = GetCharArray().textReader)
+                {
+                    Assert.Equal(0, tr.Read(new char[0], 0, 0));
```

Comment:
I assume you confirmed this failed without your fix in place.

Reply:
Yah without the change it was throwing indexOutOfRangeException as Stephen mentioned